Função para Zipar

Vinytt opened the pull request:
Histórico de alterações:

Comentar função remove_files()
Comentar ação que chamava a função remove_files() dentro da função cronjob()
Importar módulo zipfile
Criar  implementar função zip_non_used_files()

Vinytt committed:
teste commit no fork

Vinytt committed:
funcao para zipar imagens nao usadas

Vinytt committed:
Funcao para zipar, agora funcionando como deveria

gustapoll commented on main.py:
```diff
@@ -178,13 +180,23 @@ def start_cron():
+
+def zip_non_used_files():
+    for hashtag in HASHTAG_LABELS:
+        path = "./#" + hashtag
```
💅 syntax sugar
```suggestion
        path = f"./#{hashtag}"
```

gustapoll commented on main.py:
```diff
@@ -178,13 +180,23 @@ def start_cron():
+def zip_non_used_files():
+    for hashtag in HASHTAG_LABELS:
+        path = "./#" + hashtag
+        arqZip = zipfile.ZipFile("#" + hashtag + "ZIP.zip", "w", zipfile.ZIP_DEFLATED)
```
💅 syntax sugar
```suggestion
        arqZip = zipfile.ZipFile(f"#{hashtag}ZIP.zip", "w", zipfile.ZIP_DEFLATED)
```

gustapoll replied:
The option `ZIP_DEFLATED` requires the `zlib` package to be installed in order to work properly (https://docs.python.org/3/library/zipfile.html). Why it is not included in the dependency requirements? 🤔

gustapoll replied:
Also, there is a reason why we preferred `zlib` compression instead of `bz2` or `lzma`? I don't have a strong argument for or against any of them, but I was just curious to hear your thoughts on this 😄

Vinytt replied:
Oh, I'll add the package in the code. Strange it worked without it, though.
And I decided for the ZIP_DEFLATED because it was stated to be the "numeric constant for the usual ZIP compression method". I thought it would be better to go for the "usual", but I admit I don't really know the differences of each compression method hahaha

gustapoll commented on main.py:
```diff
@@ -178,13 +180,23 @@ def start_cron():
+        path = "./#" + hashtag
+        arqZip = zipfile.ZipFile("#" + hashtag + "ZIP.zip", "w", zipfile.ZIP_DEFLATED)
+        for file in os.listdir(path):
+            arqZip.write(os.path.join(path, str(file)), str(file))
```
Why do you need to cast `file` to string? 🤔

gustapoll replied:
💅  If you indeed need this cast, I would recommend to create a variable in order to make the code more clear and uncluttered.

```suggestion
            file_name = str(file)
            arqZip.write(os.path.join(path, file_name), file_name)
```

gustapoll replied:
Why do you require the second argument in `arqZip.write`? I feel like you are passing the file name twice 🤔

gustapoll commented on main.py:
```diff
@@ -178,13 +180,23 @@ def start_cron():
+        arqZip = zipfile.ZipFile("#" + hashtag + "ZIP.zip", "w", zipfile.ZIP_DEFLATED)
+        for file in os.listdir(path):
+            arqZip.write(os.path.join(path, str(file)), str(file))
+            print("Zipando " + str(file))
```
💅 syntax sugar

```suggestion
            print(f"Zipando {str(file)}")
```

gustapoll commented on main.py:
```diff
@@ -178,13 +180,23 @@ def start_cron():
+    for hashtag in HASHTAG_LABELS:
+        path = "./#" + hashtag
+        arqZip = zipfile.ZipFile("#" + hashtag + "ZIP.zip", "w", zipfile.ZIP_DEFLATED)
+        for file in os.listdir(path):
+            arqZip.write(os.path.join(path, str(file)), str(file))
+            print("Zipando " + str(file))
+        arqZip.close()
     
 if __name__ == '__main__':
     #download_hashtags_last_7_days(hashtag_labels)
     #os.system('python sort_best.py')
     #'foragarimpoforacovid', 'projetemos', 'memes_forevis' 
-    for hashtag_label in ['projetemos']:
-        download_video(hashtag_label, 10000 ,custom_data)
+    #for hashtag_label in ['projetemos']:
+    #   download_video(hashtag_label, 10000 ,custom_data)
```
Why do we commented this piece of code? 🤔

Vinytt replied:
This was part of the main, where it was changed just focus on download posts from the hashtag "#projetemos". But the only think the main should do is call "start_cron()", to begin the scheduled activities. I'll make sure to clean this part (and the others with commented code) before the next pull request!

gustapoll commented on main.py:
```diff
@@ -151,25 +152,26 @@ def cronjob():
-            if (hashtag in os.listdir('./best/' + day)): # if hashtag in this day
-                for classification in os.listdir('./best/' + day + '/' + hashtag): # for all classification in best/day/hashtag
-                    for f in os.listdir('./best/' + day + '/' + hashtag + '/' + classification): # for all files in best/day/hashtag/classification
-                        file_names.append(f) # append name in list 'file_names'
+#def remove_files():
+#    today_date = dt.date.today()
+#    last_month_date = today_date - dt.timedelta(days=30)
+#    for hashtag in HASHTAG_LABELS: # for all hashtags
+#        file_names = []
+#        for day in os.listdir('./best'):  # for every single day in 'best' folder
+#            if (hashtag in os.listdir('./best/' + day)): # if hashtag in this day
+#                for classification in os.listdir('./best/' + day + '/' + hashtag): # for all classification in best/day/hashtag
+#                    for f in os.listdir('./best/' + day + '/' + hashtag + '/' + classification): # for all files in best/day/hashtag/classification
+#                        file_names.append(f) # append name in list 'file_names'
```
We probably should reconsider this practice of leaving commented code.

gustapoll reviewed (approved):
LGTM! 🚀

Great work!! Pre-approved with some questions, suggestions and nitpicks 💅 

Also, despite not being related to this PR in particular, I would recommend some materials about git flow, since you told me that this is the first time you work with PRs and stuff (they helped me a lot back in the day 😉 )

> Git Flow: [https://medium.com/trainingcenter/utilizando-o-fluxo-git-flow-e63d5e0d5e04](url)
> Semantic Commit Messages: [https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716](url)  

P.S.: Revisei em inglês pq o código tbm está no mesmo idioma.

Vinytt committed:
Update main.py

Co-authored-by: gustapoll <59609311+gustapoll@users.noreply.github.com>

Vinytt committed:
Update main.py

Co-authored-by: gustapoll <59609311+gustapoll@users.noreply.github.com>

Vinytt committed:
Update main.py

Co-authored-by: gustapoll <59609311+gustapoll@users.noreply.github.com>

Vinytt committed:
mudancas sugeridas na main

Vinytt committed:
Funcao que adiciona os .zip no drive

Vinytt committed:
last tests in work

Vinytt committed:
final changes to the code be functional

Vinytt committed:
Only removed unecessary variables

Vinytt committed:
Update main.py

Tinha esquecido de colocar a função de pegar tweets, essa foi por pouco! Preciso prestar mais atenção...
